spyder/utils/tests/test_get_words.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+
+"""Tests for programs.py."""
+
+from os.path import dirname, abspath, splitext, join
+from spyder.utils.introspection.utils import get_words
+
+HERE = dirname(abspath(__file__))
+TEST_DATA_PATH = join(HERE, 'data')
+
+def test_get_words_html():
+    """Test for get word from html file syntax."""
+    f_in = join(TEST_DATA_PATH, "example.html")
+    words = get_words(f_in)
+    expected_words = ['DOCTYPE', 'Hello', 'Jamie', 'World', 'body', 'charset', 'en', 'h',
+                      'head', 'here', 'html', 'lang', 'meta', 'p', 'title', 'utf', 'was']
+    assert sorted(words) == sorted(expected_words)
+
+def test_get_words_R():
+    """Test for get word from R file syntax."""
+    f_in = join(TEST_DATA_PATH, "example.R")
+    words = get_words(f_in)
+    expected_words = ['Hello', 'function', 'hello', 'name', 's', 'sprintf']
+    assert sorted(words) == sorted(expected_words)
+
+def test_get_words_content_R():
+    """Test for get word from R file syntax."""
+    f_in = join(TEST_DATA_PATH, "example.R")
+    ext = splitext(f_in)[1]
+    with open(f_in, 'r') as infile:
```

Comment:
I have finally set fixtures to perform both test (file path or file content) at the same time for each file type within the test / data folder